src/core/WebAppManager.cpp

```diff
@@ -635,8 +631,11 @@ void WebAppManager::killCustomPluginProcess(const std::string& basePath)
  * @todo: this should now be moved private and be protected...leaving it for now as to not break stuff and make things
  * slightly faster for intra-sysmgr mainloop launches
  */
-std::string WebAppManager::launch(const std::string& appDescString, const std::string& params,
-        const std::string& launchingAppId, int& errCode, std::string& errMsg)
+std::string WebAppManager::launch(const std::string& appDescString,
+                                  const std::string& params,
+                                  const std::string& launchingAppId,
+                                  int& errCode, std::string& errMsg,
+                                  std::list<ShellSurface *> surfaces)
```

Comment:
It should be more clear the ownership of the ShellSurface pointers. I.e. this is stored in the end in WebAppWayland and it is not clear who owns it and when it stops owning it.

Reply:
Yes, if it is not shared, then unique_ptr so the ownership is transferred to the holder.